controllers/replica_set_controller.go

```diff
@@ -448,13 +501,24 @@ func buildAutomationConfig(mdb mdbv1.MongoDBCommunity, auth automationconfig.Aut
 
 // buildService creates a Service that will be used for the Replica Set StatefulSet
 // that allows all the members of the STS to see each other.
-// TODO: Make sure this Service is as minimal as possible, to not interfere with
-// future implementations and Service Discovery mechanisms we might implement.
-func buildService(mdb mdbv1.MongoDBCommunity) corev1.Service {
+//
+// If `isArbiter` is true, the function will create a Service suitable for the
+// Arbiter's StatefulSet.
+func buildService(mdb mdbv1.MongoDBCommunity, isArbiter bool) corev1.Service {
 	label := make(map[string]string)
-	label["app"] = mdb.ServiceName()
+	if isArbiter {
+		label["app"] = mdb.ArbiterServiceName()
+	} else {
+		label["app"] = mdb.ServiceName()
+	}
```

Comment:
[nit] to keep things as flat as possible we could do

```golang
name := mdb.ServiceName()
if isArbiter {
    name = mdb.ArbiterServiceName()
}
label["app"] = name

```

will leave it to you! This is just my personal preference.